File: backend/procmin-emdapplications/src/main/java/org/processmining/emdapplications/hfdd/data/csgraph/visualization/diffdecompgraph/DDGEdgeInterSet.java
package org.processmining.emdapplications.hfdd.data.csgraph.visualization.diffdecompgraph;

/**
 * Edge between two sets of activities that introduce a difference.
 * @author brockhoff
 *
 */
public final class DDGEdgeInterSet extends DDGEdge {
	
	/**
	 * Associated probability in the left log.
	 */
	private final double probabilityLeft;

	/**
	 * Associated probability in the right log.
	 */
	private final double probabilityRight;
	
	private boolean isEdgeInLayoutTree;

	public DDGEdgeInterSet(int id, double probabilityLeft, 
			double probabilityRight) {
		super(id);
		this.probabilityLeft = probabilityLeft;
		this.probabilityRight = probabilityRight;
		this.isEdgeInLayoutTree = false;
	}

	////////////////////////////////////////////////////////////////////////////////
	// Getter / Setter
	////////////////////////////////////////////////////////////////////////////////

	public double getProbabilityLeft() {
		return probabilityLeft;
	}

	public double getProbabilityRight() {
		return probabilityRight;
	}

	@Override
	public DDGEdgeType getEdgeType() {
		return DDGEdgeType.INTERSET;
	}

	public boolean isEdgeInLayoutTree() {
		return isEdgeInLayoutTree;
	}

	public void setEdgeInLayoutTree(boolean isEdgeInLayoutTree) {
		this.isEdgeInLayoutTree = isEdgeInLayoutTree;
	}
	
}

File: hfddFrontend/hfddApp/src/app/modules/hfdd/data/dd-graph/edge/DDGEdgeInterset.ts
import { DDGEdge } from "./DDGEdge";
import { DDGEdgeType } from "./DDGEdgeType";
import { DDGVertex } from "../vertex/DDGVertex";

export class DDGEdgeInterset extends DDGEdge {

    /**
     * Probability flow in the left log.
     */
    private readonly _probabilityLeft: number;

    /**
     * Probability flow in the right log.
     */
    private readonly _probabilityRight: number;

    /**
     * Is the edge contained in the
     * spann tree of the hierarchy?
     */
    private readonly _isSpannTreeEdge: boolean;

    public constructor(id: number, u: DDGVertex, v: DDGVertex, probabilityLeft: number, probabilityRight: number, 
            isSpannTreeEdge: boolean) {
        super(id, DDGEdgeType.INTERSET, u, v);
        this._probabilityLeft = probabilityLeft;
        this._probabilityRight = probabilityRight;
        this._isSpannTreeEdge = isSpannTreeEdge;
    }

    public calcWidth(edgeUnitProbability: number): void {
        this.width = 10;
    }

    ////////////////////////////////////////////////////////////////////////////////
    // Getter and Setter
    ////////////////////////////////////////////////////////////////////////////////
    public get probabilityLeft(): number {
        return this._probabilityLeft;
    }

    public get probabilityRight(): number {
        return this._probabilityRight;
    }

    public get isSpannTreeEdge(): boolean {
        return this._isSpannTreeEdge;
    }

}